src/main/java/eu/dissco/orchestration/backend/service/MappingService.java

```diff
@@ -42,6 +47,15 @@ public MappingRecord updateMapping(String id, Mapping mapping, String userId) {
     }
   }
 
+  public void deleteMapping(String id) throws NotFoundException {
+    int result = repository.mappingExists(id);
+    if (result > 0){
+      Instant deleted = Instant.now();
+      repository.deleteMapping(id, deleted);
```

Comment:
Feels a bit strange to have a different method here than in update for checking if the mapping exists. Might make these the same, and only check latest version (same as update). If we delete we always delete all.

Reply:
True true. This code was written like this originally, changed to `fetchOptionals` for both functions